test_python/test_c2.py

```diff
@@ -65,6 +63,42 @@ def ref(X, W):
             reference=ref,
         )
 
+    @given(n=st.integers(1, 128),
+           m=st.integers(1, 128),
+           k=st.integers(1, 128),
+           seed=st.integers(min_value=0, max_value=2**32 - 1),
+           **hu.gcs_gpu_only)
+    @settings(max_examples=2)
+    def test_matmul_tune_and_run(self, n, m, k, seed, gc, dc):
+        matmul = tc.define(MATMUL_LANG, name="matmul")
```

Comment:
this test is bit unclear to me. It looks like both pytorch and C2 pybinds are being used and doing redundant work here. Is the goal to use pytorch autotuner API to get the options and use them? If that's the case, I recommend to add a `genetic_autotuner_caffe2.h` following https://github.com/facebookresearch/TensorComprehensions/blob/master/tc/autotuner/genetic_autotuner_aten.h this should be long term stable solution and doesn't require much coding. Ping me if you want to do this.

Reply:
@prigoyal , let me actually respectfully disagree. I think this code is an example how bindings can be reused across frameworks without need to introduce new classes.